### ISSUE TITLE
Mnemonic drops to the first one after `kaspawallet new-address`

### ISSUE THREAD

**Issue by younicoin** [kaspanet/kaspad#2151]
Hello! I just compiled kaspad from source, but see the bug that my mnemonic magically restores to old one when I run kaspawallet new-address

Steps to reproduce
1. Start kaspad daemon: `kaspad --utxoindex`
2. Run kaspawallet daemon: `kaspawallet start-daemon`
3. Create a wallet and say yes to store new keys : `kaspawallet create`
4. Check mnemonic: 
```
kaspawallet dump-unencrypted-data
This operation will print your unencrypted keys on the screen. Anyone that sees this information will be able to steal your funds. Are you sure you want to proceed (y/N)? y

Password:
Mnemonic #1:
annual bomb lady ticket brick age enrich donor wonder future live alpha play away adjust lucky list rail minimum pencil enough arrow patrol pretty
```

5. Generate new address: `kaspawallet new-address`
6. Now create new wallet and say yes to replace keys.json file:  `kaspawallet create`
7. Check mnemonic, it is another:
```
kaspawallet dump-unencrypted-data
This operation will print your unencrypted keys on the screen. Anyone that sees this information will be able to steal your funds. Are you sure you want to proceed (y/N)? y

Password:
Mnemonic #1:
damage firm saddle elegant kangaroo endorse room recipe color join spray bunker always wealth cupboard glad truck arrow bounce stand strong coach wheat stay
```

8. Generate new address: `kaspawallet new-address`
9. Check mnemonic, it is the first one as it was at first time:
```
kaspawallet dump-unencrypted-data
This operation will print your unencrypted keys on the screen. Anyone that sees this information will be able to steal your funds. Are you sure you want to proceed (y/N)? y

Password:
Mnemonic #1:
annual bomb lady ticket brick age enrich donor wonder future live alpha play away adjust lucky list rail minimum pencil enough arrow patrol pretty
```

Please, I do not know, why does mnemonic drops to the first one after `kaspawallet new-address`.

**Comment by michaelsutton** [comment]
This seems like a bug with the daemon not being restarted when you recreate the wallet (overriding the keys file). In general users should restart the daemon if the wallet was overridden. Nonetheless we should protect the users from such errors. For instance by having the daemon process locking the keys file. @someone235 @svarogg what do you think ?

**Comment by michaelsutton** [comment]
Fixed by #2154
